File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> search = new HashMap<>();

        for (int k = 0; k < nums.length; k++) {
            int other = target - nums[k];
            if (search.containsKey(other)) {
                return new int[] {search.get(other), k};
            }
            search.put(nums[k], k);
        }
        return new int[] {};

    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        bottom = 0
        top = len(nums) - 1
        
        while bottom <= top:
            mid = (bottom + top) // 2
            if nums[mid] == target:
                return mid
            elif target < nums[mid]:
                top = mid - 1
            else:
                bottom = mid + 1
        return -1


        

